share-coordinator/src/test/java/org/apache/kafka/coordinator/share/ShareCoordinatorServiceTest.java

```diff
@@ -637,4 +640,35 @@ public void testTopicPartitionFor() {
         assertEquals(Topic.SHARE_GROUP_STATE_TOPIC_NAME, tp.topic());
         assertEquals(expectedPartition, tp.partition());
     }
+
+    @Test
+    public void testPartitionFor() {
+        CoordinatorRuntime<ShareCoordinatorShard, CoordinatorRecord> runtime = mockRuntime();
+        ShareCoordinatorService service = new ShareCoordinatorService(
+                new LogContext(),
+                ShareCoordinatorConfigTest.createConfig(ShareCoordinatorConfigTest.testConfigMap()),
+                runtime,
+                new ShareCoordinatorMetrics(),
+                Time.SYSTEM
+        );
+
+        String groupId = "group1";
+        Uuid topicId = Uuid.randomUuid();
+        int partition = 0;
+
+        // inactive shard should throw exception
+        assertThrows(CoordinatorNotAvailableException.class, () -> service.partitionFor(SharePartitionKey.getInstance(groupId, topicId, partition)));
+
+        final int numPartitions = 50;
+        service.startup(() -> numPartitions);
+
+        final SharePartitionKey key1 = SharePartitionKey.getInstance(groupId, new TopicIdPartition(topicId, partition, null));
+        int sharePartitionKey = service.partitionFor(key1);
+        assertEquals(Utils.abs(key1.asCoordinatorKey().hashCode()) % numPartitions, sharePartitionKey);
+
+        // The presence of a topic name should not affect the choice of partition
+        final SharePartitionKey key2 = new SharePartitionKey(groupId, new TopicIdPartition(topicId, partition, "whatever"));
+        sharePartitionKey = service.partitionFor(key2);
+        assertEquals(Utils.abs(key2.asCoordinatorKey().hashCode()) % numPartitions, sharePartitionKey);
+    }
```

Comment:
Isn't it also true that `assertEquals(key1.asCoordinatorKey(), key2.asCoordinatorKey())`? That seems like a sensible assertion too.